outlines/blaze.md

```diff
@@ -0,0 +1,56 @@
+# Blaze
+
+1. Introduction to Blaze -- the tracker-backed handlebars like templating syntax
+  1. Example of spacebars syntax, data context + helpers
+  2. Data contexts and access - name lookup (`.`, `this`, `..`, `[0]`), and null values
+  3. Helpers, arguments, options
+  4. Inclusion of other templates + arguments, passing data contexts
+  5. Helpers inside tags -- returning strings and objects, `checked={{isChecked}}`
+  6. Nested helpers + sub expressions
+  7. Block helpers (you can create them with templates, see 10. below)
+  8. Safestrings and `{{{`
+  9. Builtin block helpers
+    1. `{{#if/unless}}`
+    2. `{{#with}}`
```

Comment:
For Blaze Components I decided I will just merge lexical scope with current data context. Users should have to think about this internal implementation details. If they can write `{{foo}}` inside `each ... in` and it resolves to something, they should also be able to do `this.data().foo`. Otherwise this is really confusing.

Maybe a method on template instances like `lookup` could be helpful, which would do two things:
* go in the same order as template lookup does
* it would wrap reactivity only for that particular value

So you could do `Template.instance().lookup('foo')`. So something I am contemplating here: https://github.com/peerlibrary/meteor-blaze-components/issues/101

`resolve` could also just be a global template helper we suggest to use (same as `instance`, BTW, I have seen `instance` used more around in existing code I think).